src/scala/org/pantsbuild/zinc/compiler/Main.scala

```diff
@@ -236,9 +317,8 @@ object Main {
             .ConcreteAnalysisContents(result.analysis, result.setup)
         )
       }
-
+      dumpDiagnostics(settings, inputs, workingDirectory)
```

Comment:
nit: I'd prefer the logic of "if the user has enabled this option then do X" to be lifted here, and to keep all option-handling code to the `mainImpl` function, but that's a style preference.